afm/filesystems/vault.py

```diff
@@ -12,66 +12,73 @@ def get_jwt_from_file(file_name):
     with open(file_name) as f:
         return f.read()
 
-def vault_jwt_auth(jwt, vault_address, vault_path, role):
+def vault_jwt_auth(jwt, vault_address, vault_path, role, datasetID):
     """Authenticate against Vault using a JWT token (i.e., k8s sa token)"""
     full_auth_path = vault_address + vault_path
-    logger.debug("full_auth_path = %s", str(full_auth_path))
+    logger.trace('authenticating against vault using a JWT token',
+        extra={'full_auth_path': str(full_auth_path), DataSetID: datasetID})
     json = {"jwt": jwt, "role": role}
     response = requests.post(full_auth_path, json=json)
     if response.status_code == 200:
         return response.json()
-    logger.error("Got error code %d from Vault authentication", response.status_code)
-    logger.error("Error response: %s", str(response.json()))
+    logger.error("vault authentication failed",
+        extra={Error: str(response.status_code) + ': ' + str(response.json()),
+               DataSetID: datasetID})
     return None
```

Comment:
Please include only the error code as the value of Error.  If additional information is required put it either in the messsage or in a separate key/value pair